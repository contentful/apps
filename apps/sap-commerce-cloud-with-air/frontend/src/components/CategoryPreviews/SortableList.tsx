import * as React from 'react';
import { SortableContainer } from 'react-sortable-hoc';
import { DeleteFn, Category } from '../../interfaces';
import { SortableListItem } from './SortableListItem';

export interface Props {
  disabled: boolean;
  categoryPreviews: Category[];
  deleteFn: DeleteFn;
}

export const SortableList = SortableContainer<Props>(
  ({ disabled, deleteFn, categoryPreviews }: Props) => {
    const itemsAreSortable = categoryPreviews.length > 1;
    return (
      <div>
        {categoryPreviews.map((category, index) => {
          return (
            <SortableListItem
              disabled={disabled}
              key={`${category.id}-${category.name}`}
              category={category}
              index={index}
              onDelete={() => deleteFn(index)}
              isSortable={itemsAreSortable}
            />
          );
        })}
      </div>
    );
  }
);
