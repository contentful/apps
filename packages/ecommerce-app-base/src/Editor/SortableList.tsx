import { FC } from 'react';
import { DeleteFn, Product } from '../types';
import { SortableListItem } from './SortableListItem';

export interface Props {
  disabled: boolean;
  productPreviews: Product[];
  deleteFn: DeleteFn;
  skuType?: string;
}

export const SortableList: FC<Props> = ({ disabled, deleteFn, productPreviews = [], skuType }) => {
  const itemsAreSortable = productPreviews.length > 1;

  return (
    <div>
      {productPreviews.map((product, index) => {
        return (
          <SortableListItem
            disabled={disabled}
            key={`${product.image}-${product.sku}`}
            product={product}
            onDelete={() => deleteFn(index)}
            isSortable={itemsAreSortable}
            skuType={skuType}
          />
        );
      })}
    </div>
  );
};
