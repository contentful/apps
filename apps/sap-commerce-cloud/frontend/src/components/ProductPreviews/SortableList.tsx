import { DeleteFn, Product } from '@interfaces';
import { SortableListItem } from '@components/ProductPreviews/SortableListItem';
import { styles } from '@components/ProductPreviews/SortableList.styles';

export interface Props {
  disabled: boolean;
  productPreviews: Product[];
  deleteFn: DeleteFn;
}

export function SortableList({ disabled, deleteFn, productPreviews = [] }: Props) {
  const itemsAreSortable = productPreviews.length > 1;

  return (
    <div className={styles.container}>
      {productPreviews.map((product, index) => {
        return (
          <SortableListItem
            disabled={disabled}
            key={`${product.image}-${product.sku}`}
            product={product}
            onDelete={() => deleteFn(index)}
            isSortable={itemsAreSortable}
          />
        );
      })}
    </div>
  );
}
