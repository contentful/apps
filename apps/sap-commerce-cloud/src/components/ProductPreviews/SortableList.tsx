import { FC } from 'react';
import { DeleteFn, Product } from '../../interfaces';
import { SortableListItem } from './SortableListItem';
import { styles } from './SortableList.styles';

export interface Props {
  disabled: boolean;
  productPreviews: Product[];
  deleteFn: DeleteFn;
}

export const SortableList: FC<Props> = ({ disabled, deleteFn, productPreviews = [] }) => {
  const itemsAreSortable = productPreviews.length > 1;

  return (
    <div className={styles.container}>
      {productPreviews.map((product, index) => {
        return (
          <SortableListItem
            disabled={disabled}
            key={`${product.image}-${product.sku}`}
            product={product}
            onDelete={() => deleteFn(index)}
            isSortable={itemsAreSortable}
          />
        );
      })}
    </div>
  );
};
