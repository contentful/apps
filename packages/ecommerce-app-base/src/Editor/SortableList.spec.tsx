import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Props, SortableList } from './SortableList';
import productPreviews from '../__mocks__/productPreviews';

const defaultProps: Props = {
  disabled: false,
  productPreviews,
  deleteFn: jest.fn(),
};

const renderComponent = (props: Props) => {
  return render(<SortableList {...props} />);
};

jest.mock('react-sortable-hoc', () => ({
  SortableContainer: (x: any) => x,
  SortableElement: (x: any) => x,
  SortableHandle: (x: any) => x,
}));

describe('SortableList', () => {
  afterEach(cleanup);

  it('should render successfully', async () => {
    const { queryAllByTestId } = renderComponent(defaultProps);

    expect(queryAllByTestId('sortable-list-item')).toHaveLength(productPreviews.length);
  });
});
