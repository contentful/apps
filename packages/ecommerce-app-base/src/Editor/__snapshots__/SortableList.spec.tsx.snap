// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableList should render successfully 1`] = `
<div>
  <div>
    <div
      class="Card__Card___1_26G css-pivbrd Card__Card--padding-default___xgDF9"
      data-test-id="cf-ui-card"
    >
      <div
        class="CardDragHandle__CardDragHandle___2rqnO css-x97jm9"
        data-test-id="cf-ui-card-drag-handle"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="CardDragHandle__CardDragHandle__sr-label___3CueY helpers__sr-only___3Kv3z"
        >
          Reorder product
        </span>
      </div>
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu css-eiov7n"
        data-test-id="cf-ui-skeleton-form"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#cf-ui-skeleton-clip-0)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient-1);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="cf-ui-skeleton-clip-0"
          >
            <rect
              data-test-id="cf-ui-skeleton-image"
              height="48"
              rx="0"
              ry="0"
              width="48"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient-1"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation-2"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation-2.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation-2.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
      <div
        class="css-kx63wx"
      >
        <img
          alt="Dress Twin-Set rose"
          data-test-id="image"
          src="https://example.com/commercetools-example/products/picture-1.jpg"
          style="display: none;"
        />
      </div>
      <section
        class="css-1phcpb1"
      >
        <div
          class="Typography__Typography___1ZUfE"
          data-test-id="cf-ui-text-container"
        >
          <h1
            class="Heading__Heading___83D3K css-1xd3m0r f36-margin-bottom--m"
            data-test-id="cf-ui-heading"
          >
            Dress Twin-Set rose
          </h1>
          <h2
            class="Subheading__Subheading___2mA9j css-ksy8vh f36-margin-bottom--m"
            data-test-id="cf-ui-subheading"
          >
            M0E20130820E90Z
          </h2>
        </div>
      </section>
      <div
        class="css-mfolfb"
      >
        <a
          href="https://mc.commercetools.com/example-project/products/28a7442d/general"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </a>
        <button
          class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
          data-test-id="cf-ui-icon-button"
          type="button"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
            tabindex="-1"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--primary___3M4Il IconButton__IconButton__icon___3yZQN"
              data-test-id="cf-ui-icon"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
              />
            </svg>
            <span
              class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
            >
              Delete
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="Card__Card___1_26G css-pivbrd Card__Card--padding-default___xgDF9"
      data-test-id="cf-ui-card"
    >
      <div
        class="CardDragHandle__CardDragHandle___2rqnO css-x97jm9"
        data-test-id="cf-ui-card-drag-handle"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="CardDragHandle__CardDragHandle__sr-label___3CueY helpers__sr-only___3Kv3z"
        >
          Reorder product
        </span>
      </div>
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu css-eiov7n"
        data-test-id="cf-ui-skeleton-form"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#cf-ui-skeleton-clip-3)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient-4);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="cf-ui-skeleton-clip-3"
          >
            <rect
              data-test-id="cf-ui-skeleton-image"
              height="48"
              rx="0"
              ry="0"
              width="48"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient-4"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation-5"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation-5.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation-5.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
      <div
        class="css-kx63wx"
      >
        <img
          alt="Michael Kors – Shopper “Jet Set Travel”"
          data-test-id="image"
          src="https://example.com/commercetools-example/products/picture-2.jpg"
          style="display: none;"
        />
      </div>
      <section
        class="css-1phcpb1"
      >
        <div
          class="Typography__Typography___1ZUfE"
          data-test-id="cf-ui-text-container"
        >
          <h1
            class="Heading__Heading___83D3K css-1xd3m0r f36-margin-bottom--m"
            data-test-id="cf-ui-heading"
          >
            Michael Kors – Shopper “Jet Set Travel”
          </h1>
          <h2
            class="Subheading__Subheading___2mA9j css-ksy8vh f36-margin-bottom--m"
            data-test-id="cf-ui-subheading"
          >
            A0E2300FX102203
          </h2>
        </div>
      </section>
      <div
        class="css-mfolfb"
      >
        <a
          href="https://mc.commercetools.com/example-project/products/4e5446e8/general"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </a>
        <button
          class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
          data-test-id="cf-ui-icon-button"
          type="button"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
            tabindex="-1"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--primary___3M4Il IconButton__IconButton__icon___3yZQN"
              data-test-id="cf-ui-icon"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
              />
            </svg>
            <span
              class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
            >
              Delete
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="Card__Card___1_26G css-pivbrd Card__Card--padding-default___xgDF9"
      data-test-id="cf-ui-card"
    >
      <div
        class="CardDragHandle__CardDragHandle___2rqnO css-x97jm9"
        data-test-id="cf-ui-card-drag-handle"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          class="CardDragHandle__CardDragHandle__sr-label___3CueY helpers__sr-only___3Kv3z"
        >
          Reorder product
        </span>
      </div>
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu css-eiov7n"
        data-test-id="cf-ui-skeleton-form"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#cf-ui-skeleton-clip-6)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient-7);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="cf-ui-skeleton-clip-6"
          >
            <rect
              data-test-id="cf-ui-skeleton-image"
              height="48"
              rx="0"
              ry="0"
              width="48"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient-7"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation-8"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation-8.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation-8.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
      <div
        class="css-kx63wx"
      >
        <img
          alt="Shirt “Jenny“ Polo Ralph Lauren white"
          data-test-id="image"
          src="https://example.com/commercetools-example/products/picture-3.jpg"
          style="display: none;"
        />
      </div>
      <section
        class="css-1phcpb1"
      >
        <div
          class="Typography__Typography___1ZUfE"
          data-test-id="cf-ui-text-container"
        >
          <h1
            class="Heading__Heading___83D3K css-1xd3m0r f36-margin-bottom--m"
            data-test-id="cf-ui-heading"
          >
            Shirt “Jenny“ Polo Ralph Lauren white
          </h1>
          <h2
            class="Subheading__Subheading___2mA9j css-ksy8vh f36-margin-bottom--m"
            data-test-id="cf-ui-subheading"
          >
            M0E21300900DZN7
          </h2>
        </div>
      </section>
      <div
        class="css-mfolfb"
      >
        <a
          href="https://mc.commercetools.com/example-project/products/db3c234l/general"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </a>
        <button
          class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
          data-test-id="cf-ui-icon-button"
          type="button"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
            tabindex="-1"
          >
            <svg
              class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--primary___3M4Il IconButton__IconButton__icon___3yZQN"
              data-test-id="cf-ui-icon"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
              />
            </svg>
            <span
              class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
            >
              Delete
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
